Content.Server/Radio/Components/RadioJammerComponent.cs

```diff
@@ -9,12 +9,25 @@ namespace Content.Server.Radio.Components;
 [Access(typeof(JammerSystem))]
 public sealed partial class RadioJammerComponent : Component
 {
-    [DataField("range"), ViewVariables(VVAccess.ReadWrite)]
-    public float Range = 8f;
+    [DataDefinition]
+    public partial struct RadioJamSetting
+    {
+        [DataField(required: true)]
+        public float Wattage;
 
-    /// <summary>
-    /// Power usage per second when enabled
-    /// </summary>
-    [DataField("wattage"), ViewVariables(VVAccess.ReadWrite)]
-    public float Wattage = 2f;
+        [DataField(required: true)]
+        public float Range;
+
+        [DataField(required: true)]
+        public LocId Message = string.Empty;
+
+        [DataField(required: true)]
+        public LocId Name = string.Empty;
+    }
+
+    [DataField(required: true)]
+    public List<RadioJamSetting> Settings = new();
+
+    [DataField("selectedPowerLevel"), ViewVariables(VVAccess.ReadWrite)]
+    public byte SelectedPowerLevel = 1;
```

Comment:
Could this be a RadioJamSetting?

Reply:
then data is copied, which is bad

Reply:
Yes could! I actually started out as it being one. However, I thought that having it be a byte would be the better choice. 

1.) I need the index in the for loop in verbs, having it be a byte is very simple to implement.
2.) I couldn't figure out how to initialize it to the middle value unless I just copy and pasted all the stuff in manually which is obviously wrong!

Sadly, this doesn't work :(
```
 [DataField("selectedPowerLevel"), ViewVariables(VVAccess.ReadWrite)]
    public RadioJamSetting SelectedPowerLevel = Settings[1];
```

If you can tell me how to fix the second issue I'd totally do it!

Reply:
structs get copied so you dont want to do that anyway leave it as an index

also for map saving, it would store the power level's values instead of index so it would not work after e.g. updating range to be bigger

Reply:
Ah I see! That's was kina hoping it would just be a reference but if it just does a copy anyway then sticking with the index is definitely the right choice.